libr/bin/p/bin_mz.c

```diff
@@ -58,22 +58,20 @@ static bool check_bytes(const ut8 *buf, ut64 length) {
 		}
 
 		// Check for New Executable, LE/LX or Phar Lap executable
-		if (!memcmp (buf + new_exe_header_offset, "NE", 2) ||
-			!memcmp (buf + new_exe_header_offset, "LE", 2) ||
+		if (!memcmp(buf, "MZ", 2) && 
+			!memcmp (buf + new_exe_header_offset, "NE", 2) ||
 			!memcmp (buf + new_exe_header_offset, "LX", 2) ||
 			!memcmp (buf + new_exe_header_offset, "PL", 2)) {
-			if (!checkEntrypoint (buf, length)) {
```

Comment:
as long as this conditional is pretty long and repetitive is better to move this into a separate function and make this code more readable like this: if (knownHeader(buf+new_exe_header_offset)) {

Reply:
okay sure